src/client/Microsoft.Identity.Client/Platforms/Android/AuthenticationContinuationHelper.cs

```diff
@@ -37,7 +37,7 @@ public static void SetAuthenticationContinuationEventArgs(int requestCode, Resul
             {
                 authorizationResult = ProcessFromEmbeddedWebview(requestCode, resultCode, data);
             }
-            else if (!String.IsNullOrEmpty(data.GetStringExtra(BrokerConstants.BrokerResultV2)))
+            else if (!String.IsNullOrEmpty(data.GetStringExtra(BrokerConstants.BrokerResultV2)) || requestCode == BrokerConstants.BrokerRequestId)
```

Comment:
The issue with the cancellation error was when the user cancels the authentication, there is no "BrokerResultV2". I added an extra check to check the request code that is added to the activity when it is launched. only broker adds a request code at the moment. both the system browser and embedded webview on android use a request value of zero. @bgavrilMS do you think its safe to only check for the request code here?